### PR TITLE
Final PR

### PR DESCRIPTION
**Pull request by sobota**
Fixed split()/strip() problem. New word selection strategy. Fixed word duplicaction in glove- and minor fixes.

### CODE REVIEW

File: web/embedding.py [hunk 1]
```diff
@@ -85,38 +88,99 @@ def get(self, k, default=None):
...
-    def transform_words(self, f, inplace=False):
-        """ Tranform words in vocabulary """
-        id_map = {}
-        id_map_to_new = {}
+        if clean_words:
+            tw = tw.transform_words(partial(lambda w: w.strip(" ")), inplace=inplace, lower=lower)
+        return tw
+
+    def transform_words(self, f, inplace=False, lower=False):
+        """ Transform words in vocabulary """
```
Comment (kudkudak) [inline web/embedding.py]: Add docstring explaining prioritization.

File: web/embedding.py [hunk 2]
```diff
@@ -85,38 +88,99 @@ def get(self, k, default=None):
...
+
+        elif isinstance(self.vocabulary, OrderedVocabulary):
+            words = sorted(id_map.keys(), key=lambda x: id_map[x])
+            vectors = curr_vec[[id_map[w] for w in words]]
 
+        elif isinstance(self.vocabulary, Vocabulary):
             words = sorted(id_map.keys(), key=lambda x: id_map[x])
-            self.vectors = self.vectors[0:len(id_map)]
+            vectors = curr_vec[[id_map[w] for w in words]]
+
```
Comment (kudkudak) [inline web/embedding.py]: add else: raise NotImplementedError()

File: web/embedding.py [hunk 3]
```diff
@@ -244,34 +308,48 @@ def _from_word2vec_binary(fname):
...
                                    "\n{}".format(line_no, e))
                     continue
+
                 # We differ from Gensim implementation.
                 # Our assumption that a difference of one happens because of having a
                 # space in the word.
                 if len(parts) == layer1_size + 1:
                     word, vectors[line_no - ignored] = parts[0], list(map(np.float32, parts[1:]))
-                elif len(parts) == layer1_size + 2:
+                elif len(parts) == layer1_size + 2 and parts[-1]:
```
Comment (kudkudak) [inline web/embedding.py]: Add comment explaining why this condition (it is quite cryptic)

Comment (sobota) [reply web/embedding.py]: Done

**Pull request merged**